features/telemetry/protocols/netflow/adapter/src/main/java/org/opennms/netmgt/telemetry/protocols/netflow/adapter/netflow5/Netflow5Converter.java

```diff
@@ -28,17 +28,223 @@
+                    return Direction.EGRESS;
+                }
+                return Direction.INGRESS;
+
+            }
+
+            @Override
+            public String getDstAddr() {
+                return flowMessage.getDstAddress();
+            }
+
+            @Override
+            public Optional<String> getDstAddrHostname() {
+                return Optional.of(flowMessage.getDstHostname());
+            }
+
+            @Override
+            public Long getDstAs() {
+                return flowMessage.getDstAs();
+            }
+
+            @Override
+            public Integer getDstMaskLen() {
+                return flowMessage.getDstMaskLen();
+            }
+
+            @Override
+            public Integer getDstPort() {
+                return flowMessage.getDstPort();
+            }
+
+            @Override
+            public Integer getEngineId() {
+                return flowMessage.getEngineId();
+            }
+
+            @Override
+            public Integer getEngineType() {
+                return flowMessage.getEngineType();
+            }
+
+            @Override
+            public Long getDeltaSwitched() {
+                return flowMessage.getDeltaSwitched();
+            }
+
+            @Override
+            public Long getFirstSwitched() {
+                return flowMessage.getFirstSwitched();
+            }
+
+            @Override
+            public int getFlowRecords() {
+                return flowMessage.getNumFlowRecords();
+            }
+
+            @Override
+            public long getFlowSeqNum() {
+                return flowMessage.getFlowSeqNum();
+            }
+
+            @Override
+            public Integer getInputSnmp() {
+                return flowMessage.getInputSnmpIfindex();
+            }
+
+            @Override
+            public Integer getIpProtocolVersion() {
+                return flowMessage.getIpProtocolVersion();
+            }
+
+            @Override
+            public Long getLastSwitched() {
+                return flowMessage.getLastSwitched();
+            }
+
+            @Override
+            public String getNextHop() {
+                return flowMessage.getNextHopAddress();
+            }
+
+            @Override
+            public Optional<String> getNextHopHostname() {
+                return Optional.of(flowMessage.getNextHopHostname());
+            }
+
+            @Override
+            public Integer getOutputSnmp() {
+                return flowMessage.getOutputSnmpIfindex();
+            }
+
+            @Override
+            public Long getPackets() {
+                return flowMessage.getNumPackets();
+            }
+
+            @Override
+            public Integer getProtocol() {
+                return flowMessage.getProtocol();
+            }
+
+            @Override
+            public SamplingAlgorithm getSamplingAlgorithm() {
+                switch (flowMessage.getSamplingAlgorithm().getNumber()) {
```

Comment:
`SmaplingAlgorithm` is an enum by itself. We should use the enum variants instead of the number.